pkg/scheduler/cache/cache.go

```diff
@@ -1284,29 +1204,73 @@ func (sc *SchedulerCache) processBindTask() {
 	sc.BindTask()
 }
 
+// executePreBind executes PreBind for one bindContext
+func (sc *SchedulerCache) executePreBind(ctx context.Context, bindContext *BindContext) error {
+	executedPreBinders := make([]PreBinder, 0, len(sc.preBinderRegistry.preBinders))
+
+	sc.preBinderRegistry.mu.RLock()
+	defer sc.preBinderRegistry.mu.RUnlock()
+
+	for _, preBinder := range sc.preBinderRegistry.preBinders {
+		if preBinder == nil {
+			continue
+		}
+
+		if err := preBinder.PreBind(ctx, bindContext); err != nil {
+			// If PreBind fails, rollback the executed PreBinders
+			for i := len(executedPreBinders) - 1; i >= 0; i-- {
+				if executedPreBinders[i] != nil {
+					executedPreBinders[i].PreBindRollBack(ctx, bindContext)
```

Comment:
What if `PreBindRollBack` execute failed?

Reply:
UnReserve in kube-scheduler currently doesn't return error, so the PreBindRollBack I designed also doesn't return an error. Also it is difficult for the existing Scheduler Cache to repeat the rollback after the rollback fails